internal/state/execution.go

```diff
@@ -433,14 +437,17 @@ func buildLastCommitInfo(block *types.Block, store Store, initialHeight int64) a
 
 func extendedCommitInfo(c abci.CommitInfo, votes []*types.Vote) abci.ExtendedCommitInfo {
 	vs := make([]abci.ExtendedVoteInfo, len(c.Votes))
+	var ext []byte
 	for i := range vs {
+		if c.Votes[i].SignedLastBlock {
```

Comment:
Moreover, can we guarantee it is the case?

Reply:
I'd suggest to assert here that `c.Votes[i].Validator` is equal to `votes[i].ValidatorAddress`

Reply:
It turned out to be too difficult to ensure that `votes` were supplied in the same order as in the `CommitInfo` struct, since when a validator set's loaded from the store its validators are sorted by voting power and then address (that's how they're saved).

I refactored this now to rather construct a map of votes by validator address, so the order of votes doesn't matter to this function.